Created CI testing using Github workflow

Rob-Keys opened the pull request:
I saw the project doesn't have any CI testing, so I added a Github workflow that runs all the tests in the tests/ folder.

It uses a matrix testing strategy to run all tests with python 3.9 and 3.13, on Ubuntu 24.04. These obviously aren't all the supported python versions, but this can be easily changed to formally test for more python versions or OS versions (limited by what types of Github runners are available: [Docs](https://docs.github.com/en/actions/reference/runners/github-hosted-runners))

The runner uses an in-line python script that reads from the debian/control file using regex to install all system dependencies, and simply installs python dependencies from the requirements.txt file.

For anyone worried about the cost of Github runners, they rolled back their pricing change: [link](https://github.blog/changelog/2025-12-16-coming-soon-simpler-pricing-and-a-better-experience-for-github-actions/)

jlu5 commented on .github/workflows/tests.yml:
```diff
@@ -0,0 +1,100 @@
+        if not pkg:
+            print("ERROR: Could not find 'Package: variety' in debian/control", file=sys.stderr)
+            sys.exit(1)
+        pkg_content = content[pkg.start():]
```
This is a lot of bespoke code. Would it be easier to build the .deb package here (dpkg-buildpackage -us -uc) and then install it? This also has the small benefit of making sure the .deb actually builds properly.

Rob-Keys commented:
Hi @jlu5, thanks for taking the time to review my PR! 

Building the .deb package is a much more elegant solution, so I implemented that in my new commits. 

As for the download tests, I added an environment variable and decorators in the test files so they are skipped by default in the workflow, but still run when the tests are executed locally.

I also updated a few tests to be aligned with the recent commits, something that will be done more consistently with CI!
